Terminal.Gui/Views/TextField.cs

```diff
@@ -1196,13 +1196,12 @@ public override void PositionCursor ()
 
         int pos = _cursorPosition - ScrollOffset + Math.Min (Frame.X, 0);
         int offB = OffSetBackground ();
-        Rectangle containerFrame = SuperView?.ViewportToScreen (SuperView.Viewport) ?? default (Rectangle);
-        Rectangle thisFrame = ViewportToScreen (Viewport);
+        var thisOffset = ViewportToScreen (new (new (col, Viewport.Y), new (Viewport.Width - col, Viewport.Height - Viewport.Y)));
+        var view = Application.Current is { } ? FindDeepestView (Application.Current, thisOffset.X, thisOffset.Y) : this;
```

Comment:
FindDeepestView is an internal API and should no be used in any View subclass.

We need to find another way of fixing this that doesn't use an internal API.  I do not want FindDeepestView to be public.

Reply:
I created the `IsViewLocationVisibleInViewport` method to return a bool if the view location is within the viewport. Only through the `FindDeepestView` method is it possible to confirm whether the cursor is in a visible location in the viewport.